Assets/Tests/PlayModeTests/Scenarios/UP_UserProfile.cs

```diff
@@ -0,0 +1,410 @@
+            CountlyConfiguration config = TestUtility.CreateBaseConfig()
+                .SetRequiresConsent(false);
+            Countly.Instance.Init(config);
+            TestUtility.ValidateRQEQSize(Countly.Instance, 1, 0);
+
+            SendUserData();
+            SendUserProperty();
+            TestUtility.ValidateRQEQSize(Countly.Instance, 1, 0);
+        }
+
+        // UserProfile calls in Countly.Instance.UserProfile
+        // We initialize SDK with consent requirement and give consents, check the RQ and EQ sizes, then call public methods
+        // Methods we call should record nothing in the RQ and EQ
+        [Test]
+        public void UP_201_CR_CG_A()
+        {
+            Consents[] consent = new Consents[] { Consents.Crashes, Consents.Events, Consents.Clicks, Consents.StarRating, Consents.Views, Consents.Users, Consents.Push, Consents.RemoteConfig, Consents.Location, Consents.Feedback, Consents.Sessions };
+            CountlyConfiguration config = TestUtility.CreateBaseConfigConsent(consent);
+            Countly.Instance.Init(config);
+            TestUtility.ValidateRQEQSize(Countly.Instance, 2, 0);
+
+            SendUserData();
+            SendUserProperty();
+            TestUtility.ValidateRQEQSize(Countly.Instance, 2, 0);
+        }
+
+        // UserProfile calls in Countly.Instance.UserProfile
+        // We initialize SDK with consent requirement and give no consent, check the RQ and EQ sizes, then call public methods
+        // Methods we call should record nothing in the RQ and EQ
+        [Test]
+        public void UP_202_CR_CNG_A()
+        {
+            Consents[] consent = new Consents[] { };
+            CountlyConfiguration config = TestUtility.CreateBaseConfigConsent(consent);
+            Countly.Instance.Init(config);
+            TestUtility.ValidateRQEQSize(Countly.Instance, 2, 0);
+
+            SendUserData();
+            SendUserProperty();
+            TestUtility.ValidateRQEQSize(Countly.Instance, 2, 0);
+        }
+
+        // RecordEventAsync with UserProfile changes
+        // We record events without consent requirement, between events we record user profile data
+        // With each event, if user data information is recorded, it should flush EQ and send user profile data
+        [Test]
+        public void UP_203_CNR_A_Events()
+        {
+            CountlyConfiguration config = TestUtility.CreateBaseConfig()
+                .SetRequiresConsent(false);
+            Countly.Instance.Init(config);
+            Countly cly = Countly.Instance;
+            TestUtility.ValidateRQEQSize(cly, 1, 0);
+
+            _ = Countly.Instance.Events.RecordEventAsync("BasicEventA");
+            _ = Countly.Instance.Events.RecordEventAsync("BasicEventB");
+            SendSameData();
+            _ = Countly.Instance.Events.RecordEventAsync("BasicEventC");
+            
+            // Extract and validate user_details requests
+            TestUtility.ValidateRQEQSize(cly, 3, 1);

```

Comment:
please also comment here just simple ones like
// 1 begin session + 1 event req (A,B)+ 1 user profile

Reply:
to clarify expecteds

Reply:
adding